packages/nextjs/components/NetworksDropdown/NetworksDropdown.tsx

```diff
@@ -24,12 +24,21 @@ export const NetworksDropdown = ({ onChange }: { onChange: (options: any) => any
   const [groupedOptionsState, setGroupedOptionsState] = useState(initialGroupedOptions);
   const [selectedOption, setSelectedOption] = useState<SingleValue<Options>>(initialGroupedOptions.mainnet.options[0]);
 
-  const { addCustomChain, removeChain, resetTargetNetwork } = useGlobalState(state => ({
+  const { addCustomChain, removeChain, resetTargetNetwork, setTargetNetwork, chains } = useGlobalState(state => ({
     addCustomChain: state.addChain,
     removeChain: state.removeChain,
     resetTargetNetwork: () => state.setTargetNetwork(mainnet),
+    setTargetNetwork: state.setTargetNetwork,
+    chains: state.chains,
   }));
 
+  useEffect(() => {
+    if (selectedOption) {
+      const chain = Object.values(chains).find(chain => chain.id === selectedOption.value);
+      setTargetNetwork(chain as Chain);
+    }
+  }, [selectedOption, setTargetNetwork, chains]);
```

Comment:
Instead of having new useEffect here can we add this logic in `handleSelectChange` after this line? 

https://github.com/BuidlGuidl/abi.ninja/blob/c388d83b7bc83f0a214a553cffb1f8991f55c855/packages/nextjs/components/NetworksDropdown/NetworksDropdown.tsx#L93